src/main/java/wolox/training/client/delegate/OpenLibraryDelegate.java

```diff
@@ -0,0 +1,46 @@
+package wolox.training.client.delegate;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.HashMap;
+import java.util.Optional;
+import org.springframework.stereotype.Component;
+import wolox.training.client.feign.OpenLibraryFeignClient;
+import wolox.training.client.models.BookInfoDto;
+import wolox.training.client.models.BooksParam;
+
+@Component
+public class OpenLibraryDelegate {
+
+    private static final String ISBN_PARAM = "ISBN:";
+
+    private final OpenLibraryFeignClient openLibraryFeignClient;
+    private final ObjectMapper objectMapper;
+
+    public OpenLibraryDelegate(OpenLibraryFeignClient openLibraryFeignClient,
+        ObjectMapper objectMapper) {
+
+        this.openLibraryFeignClient = openLibraryFeignClient;
+        this.objectMapper = objectMapper;
+
+    }
+
+    public Optional<BookInfoDto> findBookByIsbn(String isbn) {
+
+        StringBuilder isbnParam = new StringBuilder();
+        isbnParam.append(ISBN_PARAM);
+        isbnParam.append(isbn);
+
+        HashMap<String, Object> bookInfoResponse = openLibraryFeignClient
+            .findBookByIsbn(new BooksParam(isbnParam.toString()));
```

Comment:
It is responsibility of the Delegate to know that it is needed to send a string with the format `ISBN:...` to the external API?  The way I see it is that this delegate should send the isbn as a parameter and when you are building the request you format it the way you need it. The thing is I don't know how this feign client works so I'm not sure if it its possible. In that case, this is ok.